app/models/manager_refresh/inventory_collection_default/infra_manager.rb

```diff
@@ -113,50 +138,125 @@ def ems_clusters(extra_attributes = {})
     def storages(extra_attributes = {})
       attributes = {
         :model_class => ::Storage,
-        :association => :storages,
+        :association => :storages
       }
 
       attributes.merge!(extra_attributes)
     end
 
     def hosts(extra_attributes = {})
       attributes = {
-        :model_class => ::Host,
-        :association => :hosts,
+        :model_class    => ::Host,
+        :association    => :hosts,
+        :builder_params => {
+          :ems_id => ->(persister) { persister.manager.id },
+        },
       }
 
       attributes.merge!(extra_attributes)
     end
 
-    def datacenter_children(extra_attributes = {})
-      datacenter_children_save_block = lambda do |_ems, inventory_collection|
-        # TODO
-      end
+    def host_storage(extra_attributes = {})
+      attributes = {
+        :model_class => ::HostStorage,
+        :association => :host_storages,
+      }
+
+      attributes.merge!(extra_attributes)
+    end
 
+    def switches(extra_attributes = {})
       attributes = {
-        :association       => :datacenter_children,
-        :custom_save_block => datacenter_children_save_block,
+        :model_class => ::Switch,
+        :manager_ref => [:uid_ems],
+        :association => :switches,
       }
 
       attributes.merge!(extra_attributes)
     end
 
-    def resource_pool_children(extra_attributes = {})
-      resource_pool_children_save_block = lambda do |_ems, inventory_collection|
-        # TODO
+    def lans(extra_attributes = {})
+      attributes = {
+        :model_class => ::Lan,
+        :manager_ref => [:uid_ems],
+        :association => :lans,
+      }
+
+      attributes.merge!(extra_attributes)
+    end
+
+    def ems_folder_children(extra_attributes = {})
+      folder_children_save_block = lambda do |ems, inventory_collection|
+        folder_collection = inventory_collection.dependency_attributes[:folders].try(:first)
+        cluster_collection = inventory_collection.dependency_attributes[:clusters].try(:first)
+        vm_collection = inventory_collection.dependency_attributes[:vms].try(:first)
+        template_collection = inventory_collection.dependency_attributes[:templates].try(:first)
+
+        dcs = folder_collection.data.select { |folder| folder.type == 'Datacenter' }
+        root = folder_collection.data.find { |folder| folder.uid_ems == 'root_dc' }
+        dcs.each do |dc|
+          uid = dc.uid_ems
+
+          clusters = cluster_collection.data.select { |cluster| cluster.datacenter_id == uid }
+          cluster_refs = clusters.map(&:ems_ref)
+
+          vms = vm_collection.data.select { |vm| cluster_refs.include? vm.ems_cluster.ems_ref }
+          templates = template_collection.data.select { |t| cluster_refs.include? t.ems_cluster.ems_ref }
+
+          ActiveRecord::Base.transaction do
+            host_folder = EmsFolder.find_by(:uid_ems => "#{uid}_host")
```

Comment:
using the dc uid + `_host` is also very ovirt specific, same lower down for the vm folder